raiden_contracts/tests/test_unit_internals.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+
+from itertools import chain, product
+
+import pytest
+from eth_tester.constants import UINT256_MIN, UINT256_MAX
+from eth_tester.exceptions import TransactionFailed
+
+from raiden_contracts.tests.fixtures import fake_bytes
+from raiden_contracts.tests.fixtures.config import fake_hex
+from web3.exceptions import ValidationError
+
+
+def test_min_uses_usigned(token_network_test):
+    """ Min cannot be called with negative values. """
+    INVALID_VALUES = [-UINT256_MAX, -1]
+    VALID_VALUES = [UINT256_MIN, UINT256_MAX, UINT256_MAX]
+
+    all_invalid = chain(
+        product(VALID_VALUES, INVALID_VALUES),
+        product(INVALID_VALUES, VALID_VALUES),
+    )
+
+    for a, b in all_invalid:
+        with pytest.raises(ValidationError):
+            token_network_test.functions.minPublic(a, b).call()
+
+
+def test_max_uses_usigned(token_network_test):
+
+    INVALID_VALUES = [-UINT256_MAX, -1]
+    VALID_VALUES = [UINT256_MIN, UINT256_MAX, UINT256_MAX]
+
+    all_invalid = chain(
+        product(VALID_VALUES, INVALID_VALUES),
+        product(INVALID_VALUES, VALID_VALUES),
+    )
+    for a, b in all_invalid:
+        with pytest.raises(ValidationError):
+            token_network_test.functions.maxPublic(a, b).call()
+
+
+def test_min(token_network_test):
+
+    VALUES = [UINT256_MIN, 1, UINT256_MAX, UINT256_MAX]
+    for a, b in product(VALUES, VALUES):
+        assert token_network_test.functions.minPublic(a, b).call() == min(a, b)
+
+
+def test_max(token_network_test):
+
+    VALUES = [UINT256_MIN, 1, UINT256_MAX, UINT256_MAX]
+    for a, b in product(VALUES, VALUES):
+        assert token_network_test.functions.maxPublic(a, b).call() == max(a, b)
+
+
+def test_recoverAddressFromWithdrawMessage(
```

Comment:
I would propose to keep signature verification in the same place. Either add tests for this in a separate file for each type of signature, or add them to the specific test file - e.g. I already have a test for `recoverAddressFromWithdrawMessage` in https://github.com/raiden-network/raiden-contracts/blob/5468fcdc249644838d21ef1e90849b6aa1273b20/raiden_contracts/tests/test_channel_withdraw.py#L17

We don't need to keep the original `raiden` order if we cannot easily track what additional tests need to be added.

Reply:
The order/grouping of the tests is currently changing. My current approach is to have tests in the `test_unit_internals` for every function in `token_network` in the same order as they appear there. Imo if ordered in this way, it will come out most clearly that every function is tested. This would imply refactored/duplicated testing of some tests that are already in place. Let me know your stance pls.

Reply:
You could also break `test_unit_internals` into `test_unit_internals_signatures` & others.  
If you see actual duplicated tests in another file, you can just remove them from there.
Otherwise, I think it is fine.